components/brave_shields/core/common/ad_block_only_mode/ad_block_only_mode_policy_provider.cc

```diff
@@ -0,0 +1,100 @@
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_shields/core/common/ad_block_only_mode/ad_block_only_mode_policy_provider.h"
+
+#include "base/values.h"
+#include "brave/components/brave_shields/core/common/brave_shield_utils.h"
+#include "brave/components/brave_shields/core/common/pref_names.h"
+#include "components/content_settings/core/common/content_settings.h"
+#include "components/policy/core/common/configuration_policy_provider.h"
+#include "components/policy/core/common/policy_bundle.h"
+#include "components/policy/core/common/policy_map.h"
+#include "components/policy/core/common/policy_namespace.h"
+#include "components/policy/core/common/policy_types.h"
+#include "components/policy/policy_constants.h"
+#include "components/prefs/pref_service.h"
+
+namespace brave_shields {
+
+namespace {
+
+void SetPolicy(policy::PolicyMap& policies,
+               std::string_view policy_key,
+               base::Value value) {
+  policies.Set(policy_key.data(), policy::POLICY_LEVEL_MANDATORY,
+               policy::POLICY_SCOPE_USER, policy::POLICY_SOURCE_BRAVE,
+               std::move(value), nullptr);
+}
+
+void SetPolicies(policy::PolicyMap& policies) {
+#if !defined(IS_IOS)
+  // Allow JavaScript globally.
+  SetPolicy(policies, policy::key::kDefaultJavaScriptSetting,
+            base::Value(ContentSetting::CONTENT_SETTING_ALLOW));
+
+  // Allow all cookies.
+  SetPolicy(policies, policy::key::kDefaultCookiesSetting,
+            base::Value(ContentSetting::CONTENT_SETTING_ALLOW));
+
+  // Do not block third-party cookies.
+  SetPolicy(policies, policy::key::kBlockThirdPartyCookies, base::Value(false));
+
+  // Disable language fingerprinting reduction.
+  SetPolicy(policies, policy::key::kBraveReduceLanguageEnabled,
+            base::Value(false));
+
+  // Disable De-AMP.
+  SetPolicy(policies, policy::key::kBraveDeAmpEnabled, base::Value(false));
+
+  // Disable URL debouncing.
+  SetPolicy(policies, policy::key::kBraveDebouncingEnabled, base::Value(false));
+#endif  // !defined(IS_IOS)
+}
+
+}  // namespace
+
+AdBlockOnlyModePolicyProvider::AdBlockOnlyModePolicyProvider(
+    PrefService* local_state,
+    policy::ConfigurationPolicyProvider& policy_provider)
+    : local_state_(local_state), policy_provider_(policy_provider) {
+  CHECK(local_state_);
+}
+
+AdBlockOnlyModePolicyProvider::~AdBlockOnlyModePolicyProvider() = default;
+
+void AdBlockOnlyModePolicyProvider::Init() {
+  if (!IsAdblockOnlyModeFeatureEnabled()) {
+    return;
+  }
+
+  pref_change_registrar_.Init(local_state_);
+  pref_change_registrar_.Add(
+      prefs::kAdBlockOnlyModeEnabled,
+      base::BindRepeating(
+          &AdBlockOnlyModePolicyProvider::OnAdBlockOnlyModeChanged,
+          weak_factory_.GetWeakPtr()));
+}
+
+void AdBlockOnlyModePolicyProvider::MaybeLoadPolicies(
+    policy::PolicyBundle& bundle) {
+  if (!IsAdblockOnlyModeFeatureEnabled()) {
```

Comment:
please avoid wrapping simple feature checks in methods, in particular when they pull in additional deps that would otherwise not be needed. This kind of method (if it exists at all) should be features.h in a separate `features` target so it doesn't pull in any more dependencies than necessary.